### DISCUSSION TITLE
How to compare variables for equality?

### DISCUSSION THREAD

**Discussion by anakadote** [statamic/cms#5222] [Troubleshooting]
In v2 I could filter form submissions by their data, but that's no longer working:

```
{{ form:submissions in="payment" transaction_id:is="{session:payment_transaction_id}" limit="1" }}
```

In v3 the above outputs all the submissions. So now I'm trying to use an `if` condition to only output the targeted submission. 
I've tried every possible way I can think of to compare the session variable against the submission variable for equality, to no avail:

```
{{ form:submissions in="payment" }}

    // Some of the variations I've tried...
    {{ if {session:payment_transaction_id} == transaction_id }}
    {{ if {session:payment_transaction_id} == {transaction_id} }}
    {{ if session:payment_transaction_id == transaction_id }}
    {{ if session:payment_transaction_id == {transaction_id} }}
    {{ if {session:payment_transaction_id} is transaction_id }}
    {{ if {session:payment_transaction_id} equals transaction_id }}
```

None of the above resolve to `true`, and some throw exceptions. Any help is appreciated!

And here's an example form submission yaml file:
```
transaction_id: test_6201982ddadd0
first_name: Test
last_name: Person
company: 'Obrien Guy Plc'
email: test@mailinator.com
phone: '(179) 449-3570'
comments: 'Est quia omnis lore'
invoice: 'Minus labore dolores'
amount: '12.33'
zip: '84428'
```

**Comment by jackmcdade** [comment]
You almost had it! Let's go over the rules so you can understand what does and doesn't work.

`session` is Tag, which means it **needs** to be wrapped in single braces: `{session:payment_transaction_id}`. On the other hand, `transaction_id` is a var, so it doesn't. `is` and `equals` aren't a thing, which leads you to...

```
{{ if {session:payment_transaction_id} == transaction_id }}
```

**Reply by anakadote** [reply]
Thanks, @jackmcdade , but that was the first one in my list of attempts above. I guess that means there's something else going on, hm...

**Reply by jackmcdade** [reply]
I feel really dumb – missed it entirely. 😅 

What does `{{ session:payment_transaction_id }}` return inside the `{{ form:submissions }}` tag?

**Reply by anakadote** [reply]
It returns the correct value (test_6201982ddadd0 in this case). I just setup a bare minimum test:
```html
{{ session:set payment_transaction_id="test_6201982ddadd0" }}

<div class="p-5">
{{ form:submissions in="payment" }}

    Testing: {{ session:payment_transaction_id }} == {{ transaction_id }}<br>

    {{ if {session:payment_transaction_id} == transaction_id }}

        YES

    {{ else }}

        NO

    {{ /if }}

{{ /form:submissions }}
</div>
```
Which outputs this:
<img width="581" alt="Screen Shot 2022-02-09 at 4 00 06 PM" src="https://user-images.githubusercontent.com/2798169/153289313-2abf3a91-6a15-48b9-b36d-e7f589719702.png">
